src/util.cc

```diff
@@ -1,8 +1,187 @@
 #include "util.h"
 #include "string_bytes.h"
 
+#define UNI_SUR_HIGH_START    0xD800UL
+#define UNI_SUR_LOW_END       0xDFFFUL
+#define UNI_REPLACEMENT_CHAR  0x0000FFFDUL
+#define UNI_MAX_LEGAL_UTF32   0x0010FFFFUL
+
+inline uint32_t log2(uint8_t v) {
+  const uint32_t r = (v > 15) << 2;
+  v >>= r;
+  const uint32_t s = (v > 3) << 1;
+  v >>= s;
+  v >>= 1;
+  return r | s | v;
+}
+
+inline uint32_t clz(uint8_t v) {
+  return 7 - log2(v);
+}
+
 namespace node {
 
+typedef size_t (*ErrorStrategy)(size_t, const uint8_t*, size_t);
+typedef void (*GlyphStrategy)(const uint8_t*, size_t, uint32_t, size_t);
+
+
+inline size_t Skip(
+    const size_t remaining,
+    const uint8_t* input,
+    const size_t glyph_size) {
```

Comment:
Style issue: the first argument goes on the same line as the function name and the other arguments should line up below it.  The only time that's deviated from is when the 80 column limit is exceeded.